lib/credit_card_sanitizer.rb

```diff
@@ -170,7 +172,22 @@ def is_tracking?(candidate, options)
   end
 
   def valid_numbers?(candidate, options)
-    LuhnChecksum.valid?(candidate.numbers) && valid_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+    LuhnChecksum.valid?(candidate.numbers) && valid_company_prefix?(candidate.numbers) && valid_grouping?(candidate, options) && !is_tracking?(candidate, options)
+  end
+
+  def valid_context?(options)
+    return true if valid_prefix?(options[:prefix]) && valid_postfix?(options[:postfix])
+    !options[:parse_flanking]
```

Comment:
Perhaps short circuit if we don't care: 

`!options[:parse_flanking] || (valid_prefix?(options[:prefix]) && valid_postfix?(options[:postfix]))`